### PR TITLE
Fix Python 3 Syntax Errors (en masse)

### PR DESCRIPTION
**Pull request by cclauss**
Like the other _en masse_ PRs, #3206 and #3220 but this one focuses on fixing Python 3 syntax errors.  Each line modified in this PR is one of the following:
* A Python 3 syntax error like old style print or old style exceptions or explicit tuples as function parameters, etc
* __from \_\_future\_\_ import print_function__  # only used where needed
* __from six.moves import xrange__  # should have been in #3206 but was hidden by a syntax error
* __from six.moves import input__
* __raw_input() --> input()__
* __TabError__
* explicit tuples as function parameters
* trailing whitespace at end of line or end of file (my editor chews these up on save)

@nealwu

### CODE REVIEW

File: research/neural_gpu/neural_gpu_trainer.py [hunk 1]
```diff
@@ -552,7 +553,7 @@ def score_beams_prog(beams, target, inp, history, print_out=False,
                 for h in history]
   tgt_set = set(target)
   if print_out:
-    print "target: ", tgt_prog
+    print("target: ", tgt_prog)
```
Comment (nealwu) [inline research/neural_gpu/neural_gpu_trainer.py]: This needs `from __future__ import print_function` above.

File: research/neural_programmer/neural_programmer.py [hunk 2]
```diff
@@ -113,9 +113,9 @@ def evaluate(sess, data, batch_size, graph, i):
                                                             graph))
     gc += ct * batch_size
     num_examples += batch_size
-  print "dev set accuracy   after ", i, " : ", gc / num_examples
-  print num_examples, len(data)
-  print "--------"
+  print("dev set accuracy   after ", i, " : ", gc / num_examples)
+  print(num_examples, len(data))
+  print("--------")
```
Comment (nealwu) [inline research/neural_programmer/neural_programmer.py]: Same here--we need `from __future__ import print_function`.

File: research/neural_gpu/neural_gpu_trainer.py (hunk 1 above)
Comment (cclauss) [reply research/neural_gpu/neural_gpu_trainer.py]: Thanks!  Done.

File: research/neural_programmer/neural_programmer.py (hunk 2 above)
Comment (cclauss) [reply research/neural_programmer/neural_programmer.py]: Thanks!  Done.

Comment (nealwu) [reply research/neural_programmer/neural_programmer.py]: Excellent!

**Pull request merged**